controllers/selfnoderemediation_controller.go

```diff
@@ -244,6 +251,27 @@ func (r *SelfNodeRemediationReconciler) Reconcile(ctx context.Context, req ctrl.
 	return result, r.updateSnrStatusLastError(snr, err)
 }
 
+func (r *SelfNodeRemediationReconciler) isConfigurationExist(ctx context.Context) (bool, error) {
+	if ns, err := utils.GetDeploymentNamespace(); err != nil {
+		r.logger.Error(err, "Failed getting snr namespace")
+		return true, err
+	} else {
+		snrConfig := &v1alpha1.SelfNodeRemediationConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      v1alpha1.ConfigCRName,
+				Namespace: ns,
+			},
+		}
+		err := r.Get(ctx, client.ObjectKeyFromObject(snrConfig), snrConfig)
+		if (err != nil && apiErrors.IsNotFound(err)) || snrConfig.DeletionTimestamp != nil {
```

Comment:
nit: the nil check isn't needed on the left side. Not nit: but on the right side, because the config can be nil. So it should be `	if apiErrors.IsNotFound(err) || err == nil && snr.GetDeletionTimestamp() != nil {`

Reply:
> I don't think it could be nil since it's already initialized (it could be empty though).

fair point, I thought it can set to nil by the Get function, but that's not possible